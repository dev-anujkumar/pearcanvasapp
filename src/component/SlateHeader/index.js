import SlateHeader from './SlateHeader.jsx';
export default SlateHeader;