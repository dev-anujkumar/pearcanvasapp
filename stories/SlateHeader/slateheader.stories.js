import React from 'react';

import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import  SlateHeader  from '../../src/component/SlateHeader/slateHeader';

storiesOf('SlateHeader', module)
.add('Slate Header', () => <SlateHeader onClick={action('clicked')} />);