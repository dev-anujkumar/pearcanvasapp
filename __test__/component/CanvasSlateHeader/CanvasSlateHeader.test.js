import React from 'react';
import SlateHeader from '../../../src/component/CanvasSlateHeader';


describe('Test-handleNavClick', () => {
    let props={
        slateType:'container-introduction' , 
        slateTitle: {
            text:'C1'}, 
        slateLockInfo:{
            isLocked: true,
            timestamp: "",
            userId: "c5Test01"
        },
        disabled : 'next'
    }
        const CanvasSlateHeader = mount(<SlateHeader {...props} />)
        let CanvasSlateHeaderInstance = CanvasSlateHeader.find('SlateHeader').instance();
        const spygetLabel = jest.spyOn(CanvasSlateHeaderInstance, 'getLabel')
        const spysetDynamicStyle = jest.spyOn(CanvasSlateHeaderInstance, 'setDynamicStyle')
        it('render SlateHeader component ', () => {
            expect(CanvasSlateHeader).toHaveLength(1);
            let CanvasSlateHeaderInstance = CanvasSlateHeader.instance(); 
            expect(CanvasSlateHeaderInstance).toBeDefined();
        })
        it('onClick Event-handleNavClick-next', () => {
            const spyhandleNavClick = jest.spyOn(CanvasSlateHeaderInstance, 'handleNavClick')
            CanvasSlateHeaderInstance.handleNavClick("next");
            expect(CanvasSlateHeaderInstance.handleNavClick).toHaveBeenCalled()
            spyhandleNavClick.mockClear()
        })
        it('Test-getLabel() -default case ', () => {        
            CanvasSlateHeaderInstance.getLabel("into");
            expect(CanvasSlateHeaderInstance.getLabel).toHaveBeenCalled()
            spygetLabel.mockClear()
        })
        it('Test-getLabel() -default case ', () => {
            CanvasSlateHeaderInstance.getLabel("assessment");
            expect(CanvasSlateHeaderInstance.getLabel).toHaveBeenCalled()
            spygetLabel.mockClear()
        })
        it('Test-setDynamicStyle', () => {
            CanvasSlateHeaderInstance.setDynamicStyle("assessment",'input-text');
            expect(CanvasSlateHeaderInstance.setDynamicStyle).toHaveBeenCalled()
            spysetDynamicStyle.mockClear()
        })
       
    })
describe('onClick Event', () => {
    const onLoadMock = jest.fn();
    const CanvasSlateHeader = mount(<SlateHeader onNavigate={onLoadMock} />)
    let CanvasSlateHeaderInstance = CanvasSlateHeader.find('SlateHeader').instance();

    it('onClick Event', () => {
        const spyhandleNavClick = jest.spyOn(CanvasSlateHeaderInstance, 'handleNavClick')
        CanvasSlateHeaderInstance.handleNavClick("back")
        expect(CanvasSlateHeaderInstance.handleNavClick).toHaveBeenCalled()
        spyhandleNavClick.mockClear()
    })
    it('navigate Event ', () => {
        const onLoadMock = jest.fn();
        const component = mount(<SlateHeader onNavigate={onLoadMock} disabled='next' />);
        expect(component.props().onNavigate).toEqual(onLoadMock);
    })
})
