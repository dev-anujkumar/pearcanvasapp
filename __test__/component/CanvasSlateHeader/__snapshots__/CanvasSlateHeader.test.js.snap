// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing SlateHeader component with props render SlateHeader component  1`] = `
<SlateHeader
  slateTitle={Object {}}
  slateType="section"
>
  <div
    className="slate-title"
  >
    <div
      className="canvas-header"
      id="canvas-header"
    >
      <div
        className="slate-header"
      >
        <label
          className="header-label"
          style={
            Object {
              "fontSize": "12px",
              "minWidth": "max-content",
            }
          }
        >
          SLATE:
        </label>
      </div>
      <div
        className="input-text"
        style={
          Object {
            "minWidth": "656px",
          }
        }
      >
        <label
          className="u-hidden"
          htmlFor="txt-input"
        />
        <input
          className="txt-input"
          disabled={true}
          placeholder="title"
          type="text"
          value=""
        />
      </div>
      <ElementButton
        labelText="P"
        onClick={[Function]}
        type="backward-nav-active"
      >
        <span
          className="btn-element backward-nav-active"
          onClick={[Function]}
        >
          <img
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </span>
      </ElementButton>
      <ElementButton
        labelText="P"
        onClick={[Function]}
        type="forward-nav-active"
      >
        <span
          className="btn-element forward-nav-active"
          onClick={[Function]}
        >
          <img
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </span>
      </ElementButton>
    </div>
  </div>
</SlateHeader>
`;

exports[`Testing SlateHeader component with props render assessment Slate  1`] = `
<SlateHeader
  slateTitle={Object {}}
  slateType="assessment"
>
  <div
    className="slate-title"
  >
    <div
      className="canvas-header"
      id="canvas-header"
    >
      <div
        className="slate-header"
      >
        <label
          className="header-label"
          style={
            Object {
              "fontSize": "12px",
              "minWidth": "max-content",
            }
          }
        >
          ASSESSMENT SLATE:
        </label>
      </div>
      <div
        className="input-text"
        style={
          Object {
            "minWidth": "585px",
          }
        }
      >
        <label
          className="u-hidden"
          htmlFor="txt-input"
        />
        <input
          className="txt-input"
          disabled={true}
          placeholder="title"
          type="text"
          value=""
        />
      </div>
      <ElementButton
        labelText="P"
        onClick={[Function]}
        type="backward-nav-active"
      >
        <span
          className="btn-element backward-nav-active"
          onClick={[Function]}
        >
          <img
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </span>
      </ElementButton>
      <ElementButton
        labelText="P"
        onClick={[Function]}
        type="forward-nav-active"
      >
        <span
          className="btn-element forward-nav-active"
          onClick={[Function]}
        >
          <img
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </span>
      </ElementButton>
    </div>
  </div>
</SlateHeader>
`;

exports[`Testing SlateHeader component with props render container-introduction Slate  1`] = `
<SlateHeader
  slateTitle={Object {}}
  slateType="container-introduction"
>
  <div
    className="slate-title"
  >
    <div
      className="canvas-header"
      id="canvas-header"
    >
      <div
        className="slate-header"
      >
        <label
          className="header-label"
          style={
            Object {
              "fontSize": "12px",
              "minWidth": "max-content",
            }
          }
        >
          INTRODUCTORY SLATE:
        </label>
      </div>
      <div
        className="input-text"
        style={
          Object {
            "minWidth": "559px",
          }
        }
      >
        <label
          className="u-hidden"
          htmlFor="txt-input"
        />
        <input
          className="txt-input"
          disabled={true}
          placeholder="title"
          type="text"
          value=""
        />
      </div>
      <ElementButton
        labelText="P"
        onClick={[Function]}
        type="backward-nav-active"
      >
        <span
          className="btn-element backward-nav-active"
          onClick={[Function]}
        >
          <img
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </span>
      </ElementButton>
      <ElementButton
        labelText="P"
        onClick={[Function]}
        type="forward-nav-active"
      >
        <span
          className="btn-element forward-nav-active"
          onClick={[Function]}
        >
          <img
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </span>
      </ElementButton>
    </div>
  </div>
</SlateHeader>
`;

exports[`Testing SlateHeader component with props render default Slate  1`] = `
<SlateHeader
  slateTitle={Object {}}
  slateType=""
>
  <div
    className="slate-title"
  >
    <div
      className="canvas-header"
      id="canvas-header"
    >
      <div
        className="slate-header"
      >
        <label
          className="header-label"
          style={
            Object {
              "fontSize": "12px",
              "minWidth": "max-content",
            }
          }
        >
          SLATE:
        </label>
      </div>
      <div
        className="input-text"
      >
        <label
          className="u-hidden"
          htmlFor="txt-input"
        />
        <input
          className="txt-input"
          disabled={true}
          placeholder="title"
          type="text"
          value=""
        />
      </div>
      <ElementButton
        labelText="P"
        onClick={[Function]}
        type="backward-nav-active"
      >
        <span
          className="btn-element backward-nav-active"
          onClick={[Function]}
        >
          <img
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </span>
      </ElementButton>
      <ElementButton
        labelText="P"
        onClick={[Function]}
        type="forward-nav-active"
      >
        <span
          className="btn-element forward-nav-active"
          onClick={[Function]}
        >
          <img
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </span>
      </ElementButton>
    </div>
  </div>
</SlateHeader>
`;

exports[`Testing SlateHeader component with props render section Slate  1`] = `
<SlateHeader
  slateTitle={Object {}}
  slateType="section"
>
  <div
    className="slate-title"
  >
    <div
      className="canvas-header"
      id="canvas-header"
    >
      <div
        className="slate-header"
      >
        <label
          className="header-label"
          style={
            Object {
              "fontSize": "12px",
              "minWidth": "max-content",
            }
          }
        >
          SLATE:
        </label>
      </div>
      <div
        className="input-text"
        style={
          Object {
            "minWidth": "656px",
          }
        }
      >
        <label
          className="u-hidden"
          htmlFor="txt-input"
        />
        <input
          className="txt-input"
          disabled={true}
          placeholder="title"
          type="text"
          value=""
        />
      </div>
      <ElementButton
        labelText="P"
        onClick={[Function]}
        type="backward-nav-active"
      >
        <span
          className="btn-element backward-nav-active"
          onClick={[Function]}
        >
          <img
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </span>
      </ElementButton>
      <ElementButton
        labelText="P"
        onClick={[Function]}
        type="forward-nav-active"
      >
        <span
          className="btn-element forward-nav-active"
          onClick={[Function]}
        >
          <img
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </span>
      </ElementButton>
    </div>
  </div>
</SlateHeader>
`;
